filament/backend/src/vulkan/platform/VulkanPlatform.cpp

```diff
@@ -331,6 +331,24 @@ VkDevice createLogicalDevice(VkPhysicalDevice physicalDevice,
     return device;
 }
 
+VulkanWorkarounds checkWorkarounds(VkPhysicalDevice device) {
+    VulkanWorkarounds workarounds {};
+    VkPhysicalDeviceDriverProperties driverProperties = {
+            .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DRIVER_PROPERTIES,
+    };
+    VkPhysicalDeviceProperties2 physicalDeviceProperties2 = {
+            .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2,
+            .pNext = &driverProperties,
+    };
+    vkGetPhysicalDeviceProperties2KHR(device, &physicalDeviceProperties2);
+    std::string_view driverName { driverProperties.driverName };
+    if (driverName.find("AMDVLK for Stadia") != std::string_view::npos) {
+        workarounds.convertSpecConstToConst = true;
```

Comment:
why not always do this? this would solve the problem where we cannot use spec constants for sizing ubos.

Reply:
I have not tested this with other drivers, only desktop nvidia and AMDVLK so far.  
IIRC sizing ubos was a pixel4 (i.e. adreno) issue?

Reply:
I've changed it to always do it. PTAL.

Reply:
no sizing UBOs is actually  a SPIRV limitation!